tests/test_daemon.cpp

```diff
@@ -375,6 +378,18 @@ struct DaemonCreateLaunchAliasTestSuite : public Daemon, public FakeAliasConfig,
     {
         EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(_))
             .WillRepeatedly(Return(fake_alias_dir.path()));
+
+        EXPECT_CALL(mock_json_utils, write_json(_, _)).WillRepeatedly([](auto&& a, auto&& b) {
+            // this try catch is necessary since propagating the error hangs the test.
```

Comment:
Hmm interesting... Did you check if that is a generic problem with the test setup or even with the daemon? 

For my understanding: why does throwing from here hang the test? How is that different from other throws?